### DISCUSSION TITLE
sqlalchemy in main dependency group, sqlalchemy[postgresql...] in production sqlalchemy[aiosqlite] in dev

### DISCUSSION THREAD

**Discussion by rafalkrupinski** [python-poetry/poetry#9454] [Q&A] [answered]
Is it possible to add different extras to different groups?

I'd like something like
```
[tool.poetry.dependencies]
sqlalchemy = { version="^2.0.16", extras=["asyncio"]}

[tool.poetry.group.dev.dependencies]
sqlalchemy = {version= "^2.0.16", extras=["aiosql"]}

[tool.poetry.group.production.dependencies]
sqlalchemy = {version= "^2.0.16", extras=["postgresql-psycopgbinary"]}
```

The above setup causes this cryptic message:
```shell
poetry lock --no-update

The Poetry configuration is invalid:
  - data.group.dev.dependencies.sqlalchemy must be valid exactly by one definition (0 matches found)
```

---
edit:

After changing `extras="aiosql"` to `extras=["aiosql"]` (thanks @radoering) the message is gone, but I still can't separate the dependencies.

`poetry show --with=production` and
`poetry show --with=dev`
show exactly the same list with both aiosqlite and postgres.

**Comment by radoering** [comment, marked as answer]
> ```
> [tool.poetry.group.dev.dependencies]
> sqlalchemy = {version= "^2.0.16", extras="aiosql"}
> ```

You forgot the square brackets. `extras` must not be a string.

**Reply by rafalkrupinski** [reply]
Thank you, that solves the message, but not the problem. I'll edit my question.

**Comment by rafalkrupinski** [comment]
I needed 
`poetry show --without=production --with=dev` and
`poetry show --with=production --without=dev`
to compare dependency lists
